pkgs/build-support/mitm-cache/default.nix

```
@@ -0,0 +1,34 @@
+{ lib
+, fetchFromGitHub
+, callPackage
+, rustPlatform
+, substituteAll
+, openssl
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "mitm-cache";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "chayleaf";
+    repo = "mitm-cache";
+    rev = "6d43c4875f29c1abe21e665f15734ac8cd230c49";
+    hash = "sha256-Yrgu+ILaIuoYlM3P/+Osn2nDxr9klBzQF/098CYiGEg=";
+  };
+
+  cargoHash = "sha256-/H9Ls50fLnYb+S8dFKXPtDBDvrA305ezYZTQsAgGq3k=";
```

Was able to compile for `aarch64-darwin` after adding the `Security` framework

```suggestion
{ lib
, stdenv
, fetchFromGitHub
, callPackage
, rustPlatform
, substituteAll
, openssl
, Security
}:

rustPlatform.buildRustPackage {
  pname = "mitm-cache";
  version = "0.1.0";

  src = fetchFromGitHub {
    owner = "chayleaf";
    repo = "mitm-cache";
    rev = "6d43c4875f29c1abe21e665f15734ac8cd230c49";
    hash = "sha256-Yrgu+ILaIuoYlM3P/+Osn2nDxr9klBzQF/098CYiGEg=";
  };
  
  buildInputs = lib.optionals stdenv.isDarwin [
    Security
  ];

  cargoHash = "sha256-/H9Ls50fLnYb+S8dFKXPtDBDvrA305ezYZTQsAgGq3k=";
```